### ISSUE TITLE
 After dynamically changing "lines.effect.symbol", the effect disappears

### ISSUE THREAD

**Issue by zmzimpl** [apache/incubator-echarts#13205]
I’m sorry I can’t create issues through `echarts-issues-helper` , because my office environment does not allow access to github.io.

The problem is that I want to change `lines.effect.symbol` dynamic, but the effect disappear after I change the symbol.

Reproduce link: [demo](https://gallery.echartsjs.com/editor.html?c=x9KMQuHbKl&v=2)

I changed the symbol through the following code:

```
        myChart.on('timelinechanged', (param) => {
            switch(param.currentIndex) {
                case 0:
                    optionXyMap01.options.forEach(item => {
                        item.series[2].effect.symbol = 'circle';
                    })
                    break;
                case 1:
                    optionXyMap01.options.forEach(item => {
                        item.series[2].effect.symbol = 'arrow';
                    })
                    break;
                case 2:
                     optionXyMap01.options.forEach(item => {
                        item.series[2].effect.symbol = 'rect';
                    })
                    break;
            }
            myChart.setOption(optionXyMap01);
        })

```

**Comment by plainheart** [comment]
What version of ECharts you are using? You could try the version of 4.9.0 and feel free to give us feedback again if it still exists.

**Comment by donotlb** [comment]
@zmzimpl  I had the same problem, and i solved it by calling `chart.clear()` before calling the `setOption` method

**Comment by zmzimpl** [comment]
> @zmzimpl I had the same problem, and i solved it by calling `chart.clear()` before calling the `setOption` method

Thanks, I found it work perfect now, maybe this problem has been fixed in the version of 4.9.0? I almost forgot this issue, haha.

> What version of ECharts you are using? You could try the version of 4.9.0 and feel free to give us feedback again if it still exists.

Sorry to reply so late, maybe this problem has been fixed.

**Comment by freesou2** [comment]
> @zmzimpl I had the same problem, and i solved it by calling `chart.clear()` before calling the `setOption` method

it works, thanks
